### PR TITLE
feat: Implement voting logic

### LINKED ISSUE #12: Feature: Implement voting logic for repos
### Type of feature

Feature

### Current behavior

Today the api is listing repos based on relevancy in open sauced. 

- api.opensauced.pizza
- The above is powered by a private DB hosted on [supabase](https://supabase.com/)

### Desired solution

We need a global state that can store votes from users

- Add local state to fetch vote data
- Implement vote column in supabase db
- Alternatively we could use a public JSON file to store votes as an intermediary.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by bdougie**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

This PR adds a connection to the supabase votes column in the recommendation in order to make live updates. 

## Related Tickets & Documents

fixes #12 

## Mobile & Desktop Screenshots/Recordings

Screen Shot 2022-01-05 at 5 31 44 PM

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 📜 contributing.md
- [ ] 📓 docs
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

## [optional] Are there any post-deployment tasks we need to perform?

1. Implement supabase studio https://github.com/open-sauced/hot/issues/68
2. Make a decision on Context API vs local storage - https://github.com/open-sauced/hot/issues/38

### CODE REVIEW

**Comment by bdougie** [comment]
Still a work in progress, this needs to be applied to the PostList too. Also need to still clean up a lot of code.

File: src/components/PostGrid.jsx [hunk 1]
```diff
@@ -1,9 +1,40 @@
...
+async function fetchVotesByRepo(repoName) {
+  const { data: recommendations, error } = await supabase
+    .from('recommendations')
+    .select('votes')
+    .eq('repo_name', repoName);
+
+    console.error(error);
+
+    return recommendations[0].votes ? recommendations[0].votes : 0;
+}
```
Comment (bdougie) [inline src/components/PostGrid.jsx:18]: Should create supbase.js file to house these crud operations and make them reuable.

**Comment by bdougie** [comment]
Found that the voting data is not updated when filtering. I believe this is also due to the lack of a localstorage solution. 

Before "discussed" filter applied

Screen Shot 2022-01-05 at 7 51 39 PM

After

Screen Shot 2022-01-05 at 7 51 33 PM



Not the votes are they despite the repo being different. The votes should be fetched again when filter is applied.

**Comment by hokagedemehin** [comment]
Hi @bdougie 

I think the discussed link filters the data based on issues count, could you please explain how the voting logic is related to this filter

From what I understand, voting on each card should be the number of users that have upvoted a particular Repo if you will allow me to call each card that. so when you click on the upvote nav link it should filter the cards based on the votes from highest to lowest.

How does that relate with the filtering implemented by other nav links

File: src/lib/database.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { createClient } from '@supabase/supabase-js';
+
+// probably should move these to an env.
```
[the remark this answers is not in the document]
Comment (0-vortex) [reply src/lib/database.js:3]: no need for that level of complication, just create a `.env` with these development variables set, the build process can take them from GitHub secrets when we release/deploy. 
Also, 75% of open source projects are using dotenv wrong, read more about this here: https://12factor.net

**Comment by bdougie** [comment]
> I think the discussed link filters the data based on issues count, could you please explain how the voting logic is related to this filter

@hokagedemehin that is correct. Discussed is issues, but votes are new and not represented in the API just yet, but the challenge at the moment if votes will need to be dynamic and not static like the other data. 

For example: The stars might not change that frequent, but votes can change instantly and that can't be reflected without #38. 

As far as this repo goes, once I think I will go ahead and merge without creating .env just so I don't introduce a breaking change to other contributors. 

I also need to hold off merging until #51 is complete since we are touching the same files.

**Comment by hokagedemehin** [comment]
> I also need to hold off merging until #51 is complete since we are touching the same files.

I think that PR is closed and he did #63 to replace it

File: package.json [hunk 3]
```diff
@@ -57,8 +57,8 @@
     "tailwindcss": "^3.0.7"
   },
   "dependencies": {
+    "@supabase/supabase-js": "^1.24.0",
```
Comment (kiwicopple) [inline package.json:60]: 🌶️

**Pull request merged**